### DISCUSSION TITLE
Redirect first then dispatch

### DISCUSSION THREAD

**Discussion by dharen008** [livewire/livewire#6896] [1. Help]
`// Redirect to the register page with the created ID
		return $this->redirect( '/here/' . $thisHere->id . '/edit' );

		// Notify the user that it is already been updated
		$this->dispatch(
			'toast',
			message: 'Here successfully updated',
			title: 'Success'
		);`
		
how can i redirect first then dispatch? i need to use dispatch as much as possible to use my dynamic toast script

**Comment by danie-ramdhani** [comment]
use session. see https://livewire.laravel.com/docs/redirecting#flash-messages

**Comment by dharen008** [comment]
`@if( session( 'notification' ) )
					<script>
						document.addEventListener( 'DOMContentLoaded', () => {
							var status  = '{{ session( "notification.status" ) }}';
							var message = '{{ session( "notification.message" ) }}';
							var title   = '{{ session( "notification.title" ) }}';
							showToast( status, message, title );
						} );
					</script>
				@endif`
				
Yes, i tried that and it is working im just curious if there is something like

return $this->redirect( '/here/' . $thisHere->id . '/edit' )->dispatch();

to make it easier

**Comment by thiagomeloo** [comment]
I would also like some feature that makes this possible.

something like:

```php

    return $this->redirect('/posts')
        ->afterDispatch('toast',
            message: 'Here successfully updated',
            title: 'Success'
        );

```

**Comment by pranayb2773** [comment]
@dharen008 I had a same problem. I want to dispatch notification after redirect to another page. I am using this code. I have put in common layout file which used by all livewire component. 

```
@if(session()->has('notify'))
    <div x-data="{
        init () {
            this.$nextTick(() => {
                this.$dispatch('notify', {{ json_encode([session('notify')]) }});
            })
        }
    }"></div>
@endif
```

In livewire component,

```
// Dispatch event via session
session()->flash('notify', [
   'type' => 'success',
   'content' => 'User Created Successfully.'
]);

$this->redirectRoute('admin.users.create');
```
